base/regex.jl

```diff
@@ -164,19 +164,28 @@ See [`keys`](@ref keys(::RegexMatch)) for more information.
 
 # Examples
 ```jldoctest
-julia> m = match(r"(?<hour>\\d+):(?<minute>\\d+)(am|pm)?", "11:30 in the morning")
+julia> m = match(r"(?<hour>\\d+):(?<minute>\\d+)(am|pm)?", "The time is 11:30")
 RegexMatch("11:30", hour="11", minute="30", 3=nothing)
 
-julia> hr, min, ampm = m;
+julia> m.match
+"11:30"
 
-julia> hr
-"11"
+julia> m.captures
+3-element Vector{Union{Nothing, SubString{String}}}:
+ "11"
+ "30"
+ nothing
+
+julia> m.captures[2]    # same as m[2]
```

Comment:
Better to show `m[2]` then and leave this out.

Reply:
The docstring *stresses* about one having to access named captured groups with an index on the entire match (but not on the `captures` field), so I think its good to show where there is a difference with numbered captured groups being accessed with the `captures` field.

> The match field stores the substring of the entire matched string. The
  captures field stores the substrings for each capture group, indexed by
  number. To index by capture group name, the entire match object should be
  indexed instead, as shown in the examples.

Reply:
You can access them numbered with `m[2]` too. I think since you are showing `m.captures` and the result is a vector that implies you can index it with numbers.